src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -1073,8 +1073,14 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
         }
         
         // if specified, also include Address Type
-        if (strAddrType != null) 
+        if (strAddrType != null) {
             addr.Address_Type__c = strAddrType;
+        } else {
+            // A null value will force a newly created Address record to use its defined default value for
+            // the picklist, which if not reflected here could cause isAddressChanged() to return true
+            // in cases when it should not.
+            addr.Address_Type__c = defaultAddressType;
```

Comment:
Good question. Definitely related, but not already fixed by that PR. 

In the method a new address SObject is created for `addrNew` by calling the `ADDR_Addresses_TDTM.copyAddressStdSObjAddr` method. That method returns a null value for AddressType if `con.npe01__Primary_Address_Type__c` is null - which it would be that early in the insert process. Your new code sets `con.npe01__Primary_Address_Type__c` to the value retrieved from the Address record, but that happens later in the flow.